app/Http/Controllers/MembersController.php

```diff
@@ -269,4 +269,25 @@ public function search(\Illuminate\Http\Request $request)
 
 		return view('members.search', compact('courseList', 'courseCodes', 'courses', 'vog', 'members', 'level'));
 	}
+
+	# Search members by skills
+	public function searchSkills(\Illuminate\Http\Request $request)
+	{
+		$this->authorize("viewAny", Member::class);
+
+		$skills = $request->input('skills', []);
+		$require_how = $request->input('require_how', 'one');
+
+		$all_skills = Skill::pluck('tag', 'id')->toArray();
+
+		$members = Member::all()->filter(function ($m, $key) use ($skills, $require_how) {
```

Comment:
I think this can be done more efficient with some sql or eloquent methods. Iterating all members and their skills seem what excessive and doesn't scale well.

Reply:
I've looked into using more Eloquent but couldn't really find what I wanted. I can imagine raw SQL working but that's not my strong suit.

Reply:
Hmmm just found something that will help I think: `whereHas` (doc)[https://laravel.com/docs/7.x/eloquent-relationships#querying-relationship-existence]

Reply:
I will try some things next week. My gut feeling says it is possible. But I fear for bad performance on production with this approach.
It should be possible to get members `with`  skills and filter the skills with where

Reply:
Trying some now with `whereHas`

Reply:
I think I have a very nice solution for the query now using only Eloquent. I still can't really get the layout to do what I want though 😞